scripts/migrate_unconfirmed_valid_users.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Script to migrate users with a valid date_last_login but no date_confirmed."""
+
+import sys
+import logging
+
+from website.app import init_app
+from website.models import User
+from scripts import utils as script_utils
+from tests.base import OsfTestCase
+from tests.factories import UserFactory
+from modularodm import Q
+import datetime as dt
+
+logger = logging.getLogger(__name__)
+
+def do_migration(records):
+    for user in records:
+        user.date_confirmed = user.date_last_login
+        if not user.is_registered:
+            user.is_registered = True
+        logger.info('Finished migrating user {0}'.format(user._id))
+
+def get_targets():
+    return User.find(Q('date_confirmed', 'eq', None) & Q('date_last_login', 'ne', None))
+
+def main():
+    init_app(routes=False)  # Sets the storage backends on all models
+    if 'dry' in sys.argv:
+        for user in get_targets():
+            print(user)
+    else:
+        do_migration(get_targets())
+
+class TestMigrateNodeCategories(OsfTestCase):
+
+    def test_get_targets(self):
+        test = User.find(Q('date_confirmed', 'ne', None) & Q('date_last_login', 'ne', None))
+        assert test is not None
```

Comment:
This should test should test that the targets are correct. Set up some fixtures, e.g. a user that has `date_confirmed=None` and `date_last_login != None`, a valid user, etc., and make sure that `get_targets` returns the expected value.